core/src/test/java/org/elasticsearch/index/analysis/OpenRefineFingerprintAnalyzerTests.java

```diff
@@ -0,0 +1,68 @@
+package org.elasticsearch.index.analysis;
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.elasticsearch.test.ESTokenStreamTestCase;
+
+public class OpenRefineFingerprintAnalyzerTests extends ESTokenStreamTestCase {
+
+    public void testFingerprint() throws Exception {
+        Analyzer a = new OpenRefineAnalyzer(' ', 255, false);
+        assertAnalyzesTo(a, "foo bar@baz Baz $ foo foo FOO. FoO",
+            new String[]{"bar baz foo"});
+    }
+
+    public void testReusableTokenStream() throws Exception {
+        Analyzer a = new OpenRefineAnalyzer(' ', 255, false);
+        assertAnalyzesTo(a, "foo bar baz Baz foo foo FOO. FoO",
+            new String[]{"bar baz foo"});
+        assertAnalyzesTo(a, "xyz XYZ abc 123.2 abc",
+            new String[]{"123.2 abc xyz"});
+    }
+
+    public void testAsciifolding() throws Exception {
+        Analyzer a = new OpenRefineAnalyzer(' ', 255, false);
+        assertAnalyzesTo(a, "gödel escher bach",
+            new String[]{"bach escher godel"});
+    }
+
+    public void testPreserveOriginal() throws Exception {
+        Analyzer a = new OpenRefineAnalyzer(' ', 255, true);
+        assertAnalyzesTo(a, "gödel escher bach",
+            new String[]{"bach escher godel", "bach escher gödel"});
+    }
+
+    public void testLimit() throws Exception {
+        Analyzer a = new OpenRefineAnalyzer(' ', 3, false);
+        assertAnalyzesTo(a, "e d c b a",
+            new String[]{});
```

Comment:
Just reminding myself here - the rationale is that if combined output token length exceeds max_output_size we emit no tokens rather than some truncated form?